okhttp/src/main/kotlin/okhttp3/internal/concurrent/TaskRunner.kt

```diff
@@ -76,7 +92,7 @@ class TaskRunner(
               // If the task is crashing start another thread to service the queues.
               if (!completedNormally) {
                 lock.withLock {
```

Comment:
Is it possible to remove the lock usage in favor of a single `AtomicInteger` which tracks outstanding threads requested? `startAnotherThread` logic could then use CAS semantics to decrement if positive and only start a thread if the value was modified.